compiler/fir/checkers/src/org/jetbrains/kotlin/fir/analysis/checkers/declaration/FirKClassWithIncorrectTypeArgumentChecker.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir.analysis.checkers.declaration
+
+import org.jetbrains.kotlin.descriptors.Visibilities
+import org.jetbrains.kotlin.fir.FirFakeSourceElementKind
+import org.jetbrains.kotlin.fir.FirSession
+import org.jetbrains.kotlin.fir.analysis.checkers.context.CheckerContext
+import org.jetbrains.kotlin.fir.analysis.checkers.isSubtypeOfAny
+import org.jetbrains.kotlin.fir.analysis.diagnostics.DiagnosticReporter
+import org.jetbrains.kotlin.fir.analysis.diagnostics.FirErrors
+import org.jetbrains.kotlin.fir.analysis.diagnostics.reportOn
+import org.jetbrains.kotlin.fir.declarations.*
+import org.jetbrains.kotlin.fir.resolve.diagnostics.ConeTypeParameterInQualifiedAccess
+import org.jetbrains.kotlin.fir.resolve.inference.isKClassType
+import org.jetbrains.kotlin.fir.types.*
+
+// FE1.0 [KClassWithIncorrectTypeArgumentChecker]
+object FirKClassWithIncorrectTypeArgumentChecker : FirFileChecker() {
+    override fun check(declaration: FirFile, context: CheckerContext, reporter: DiagnosticReporter) {
+        for (topLevelDeclaration in declaration.declarations) {
+            if (topLevelDeclaration is FirCallableMemberDeclaration<*>) {
+                checkTopLevelDeclaration(topLevelDeclaration, context, reporter)
+            }
+        }
+    }
+
+    private fun checkTopLevelDeclaration(
+        declaration: FirCallableMemberDeclaration<*>,
+        context: CheckerContext,
+        reporter: DiagnosticReporter
+    ) {
+        if (declaration.visibility == Visibilities.Local) return
+
+        // prevent duplicate reporting
+        if (declaration is FirPropertyAccessor) return
```

Comment:
Is this still needed? Don't think you can have an accessor as top-level in a file.

Reply:
Yes, needed: https://github.com/JetBrains/kotlin/blob/master/compiler/testData/diagnostics/tests/declarationChecks/kClassInSignature.kt#L7

Reply:
In that case, isn't it the `FirProperty` that is top-level? Not the accessor.